t/t-batch-transfer.sh

```diff
@@ -165,6 +165,53 @@ begin_test "batch transfers with ssh endpoint (git-lfs-authenticate)"
 )
 end_test
 
+assert_ssh_transfer_session_counts() {
+  local log="$1"
+  local msg="$2"
+  local min="$3"
+  local max="$4"
+
+  local count="$(grep -c "$msg" "$log")"
+
+  [ "$max" -ge "$count" ]
+  [ "$min" -le "$count" ]
+}
+
+assert_ssh_transfer_sessions() {
+  local log="$1"
+  local direction="$2"
+  local num_objs="$3"
+  local objs_per_batch="$4"
+
+  local min_expected_start=1
+  local max_expected_start=$(( num_objs > objs_per_batch ? objs_per_batch : num_objs ))
+  local min_expected_end=1
+  local max_expected_end="$max_expected_start"
+
+  local expected_ctrl=1
+
+  # On upload we currently spawn one extra control socket SSH connection
+  # to run locking commands and never shut it down cleanly, so our expected
+  # start counts are higher than our expected termination counts.
```

Comment:
> I assume we do not create the extra control socket if locking is disabled, right? (via `lfs.locksverify=false`)

That's correct, I believe.

> Would you consider it a bug that the locking command session never shuts down cleanly?

It's not ideal, but it's not causing any harm, so far as I know.  I wrote up this issue in a bit more detail at the end of the description of #5880:

> Ultimately, we may want to avoid trying to shut down our SSH connections at the end of individual batch operations, and do some analysis of the preferred lifetime for them.  (Among other issues, when pushing, we create two separate `ControlMaster=yes` connections, one for locking operations and another for batch operations; it would be nice to avoid this duplication.)
>
> Shutting down SSH connections cleanly at the end of the entire Git LFS process may also entail some refactoring of how we cause the program to exit.  Right now, we make a token effort at [closing](https://github.com/git-lfs/git-lfs/blob/8b2882b7b791635d5deefebcc4edaa9e8eef638a/commands/run.go#L155) resources associated with our API clients, but really that just amounts to [closing](https://github.com/git-lfs/git-lfs/blob/8b2882b7b791635d5deefebcc4edaa9e8eef638a/lfshttp/client.go#L205) our logger, and when we call `os.Exit()` in several dozen places, that step is skipped and we let Go take care of closing open I/O handles.  If we want to always gracefully close multiple open SSH connections by [sending](https://github.com/git-lfs/git-lfs/blob/8b2882b7b791635d5deefebcc4edaa9e8eef638a/ssh/protocol.go#L39) a `quit` message, though, we may have to refactor these `os.Exit()` calls into a common exit routine.

It's also why I drew myself a class diagram in https://github.com/git-lfs/git-lfs/issues/5880#issuecomment-2404144414.

To manage the lifetime of all the SSH sessions, including the one used for lock verification, we may have to do some more substantial revisions throughout the code in our `commands` package to replace all the `os.Exit()` calls with another mechanism.  At any rate, it seemed out of scope for this PR.